juniper_official/Solutions/Icmp-outlier/icmp.rule

```diff
@@ -0,0 +1,252 @@
+                description "Sets health based on the RTT";
+                frequency 60s;
+                /*
+                 * Sets color to red and sends out an anomaly notification
+                 * when the response time ($rtt-average) is more than static
+                 * threshold ($rtt-threshold).
+                 */
+                term is-rtt-fine {
+                    when {
+                        greater-than-or-equal-to "$rtt-average" "$rtt-threshold";
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "Round trip time($rtt-average μs) to $host is above static threshold";
+                        }
+                    }
+                }
+                /*
+                 * Sets color to yellow and sends out an anomaly notification
+                 * when the response time ($rtt-average) is more than dynamic
+                 * threshold ($dt-response-time).
+                 */
+                term is-rtt-medium {
+                    when {
+                        equal-to "$dt-response-time" 1;
+                    }
+                    then {
+                        status {
+                            color yellow;
+                            message "Round trip time($rtt-average μs) to $host is above dynamic threshold";
+                        }
+                    }
+                }
+                /*
+                 * Defaults color to green.
+                 */
+                term rtt-normal {
+                    then {
+                        status {
+                            color green;
+                            message "Round trip time($rtt-average μs) to $host is normal";
+                        }
+                    }
+                }
+            }
+            variable count_var {
+                value 1;
+                type sensor-argument;
+                description "Input ICMP probe count";
+            }
+            variable host_var {
+                value "[null]";
+                type sensor-argument;
+                description "Input Destination host";
+            }
+            variable rtt-threshold-var {
+                value 1000;
+                description "Input RTT response static threshold in μs";
+                type int;
+            }
+        }
+        rule check-outlier {
+            description "Detects ICMP outlier response time and notify anomalies";
+            synopsis "ICMP outlier analyzer";
+            rule-frequency 60s;
+            network-rule;
+            field rtt-ol {
+                formula {
+                    outlier-detection {
+                        algorithm {
+                            dbscan {
+                                sensitivity {
+                                    level low;
+                                }
+                                learning-period 120s;
```

Comment:
The learning period is based on one ping then.. Should we make this 5x of the sensor value?

Reply:
Modified